src/vs/workbench/browser/parts/titlebar/titlebarPart.ts

```diff
@@ -182,10 +182,12 @@ export class BrowserTitleService extends MultiWindowParts<BrowserTitlebarPart> i
 	private variables: ITitleVariable[] = [];
 
 	registerVariables(variables: ITitleVariable[]): void {
-		this.variables.push(...variables);
+		const unregisteredVariables = variables.filter(v => !this.variables.some(v2 => v2.contextKey === v.contextKey));
```

Comment:
I wonder if instead of filtering we should throw or log an error when we recognize that a variable has been registered multiple times

Reply:
Yeah, we should log some info to tell the developer what is going wrong, should I use ILogService here? 
Or perhaps we can remove this code cause in [windowTitle.ts#L255](https://github.com/microsoft/vscode/blob/d751e4324d12370e950f8cd031aa8829637ce300/src/vs/workbench/browser/parts/titlebar/windowTitle.ts#L255) already do the checking?